test/bazelify_config_test.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:test/test.dart';
+
+import 'package:bazel/src/bazelify/bazelify_config.dart';
+import 'package:bazel/src/bazelify/build.dart';
+import 'package:bazel/src/bazelify/pubspec.dart';
+
+void main() {
+  test('bazelify.yaml can be parsed', () {
+    var pubspec = new Pubspec.parse(pubspecYaml);
+    var bazelifyConfig = new BazelifyConfig.parse(pubspec, bazelifyYaml);
+    expectDartLibraries(bazelifyConfig.dartLibraries, {
+      'a': new DartLibrary(
+          name: 'a',
+          package: 'example',
+          dependencies: ['b', 'c:d'],
+          sources: ['lib/a.dart', 'lib/src/a/**']),
+      'e': new DartLibrary(
+          name: 'e',
+          package: 'example',
+          dependencies: ['f'],
+          sources: ['lib/e.dart', 'lib/src/e/**'],
+          isDefault: true),
+    });
+  });
+}
+
+var bazelifyYaml = '''
+targets:
+  a:
+    dependencies:
+      - b
+      - c:d
+    sources:
+      - "lib/a.dart"
+      - "lib/src/a/**"
+  e:
+    default: true
+    dependencies:
+      - f
+    sources:
+      - "lib/e.dart"
+      - "lib/src/e/**"
+''';
+
+var pubspecYaml = '''
+name: example
+dependencies:
+  a: 1.0.0
+  b: 2.0.0
+''';
+
+void expectDartLibraries(
+    Map<String, DartLibrary> actual, Map<String, DartLibrary> expected) {
+  expect(actual.keys, unorderedEquals(expected.keys));
+  for (var p in actual.keys) {
+    expect(actual[p], equals(new _DartLibraryMatcher(expected[p])));
```

Comment:
the `equals` isn't necessary here - and is actually making the intention less clear IMO

I'm actually a little surprised it works since I'd think it would make sure the value is itself a Matcher...